spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/JdbcAggregateJobQueryMariadbDaoTests.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.batch;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MariaDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import org.springframework.cloud.dataflow.core.database.support.DatabaseType;
+
+
+@Testcontainers
+public class JdbcAggregateJobQueryMariadbDaoTests extends AbstractJdbcAggregateJobQueryDaoTests{
+
+	@Container
+	private static final JdbcDatabaseContainer dbContainer = new MariaDBContainer("mariadb:10.9.3");
+
+	@BeforeEach
+	void prepareForTest() throws Exception {
+		super.prepareForTest(dbContainer, "mariadb", DatabaseType.MYSQL);
```

Comment:
So I figured out why this has to be MYSQL for MariaDB incrementer factory... it is because on older version of MariaDB, the sequence is a table and the `org.springframework.cloud.dataflow.core.database.support.MultiSchemaIncrementerFactory#getIncrementerType` therefore will treat it as such and ultimately drop into the old incrementer factory which knows nothing about MariaDB and therefore must be MYSQL. 

I have created the 
```java
	protected String determineDatabaseType(DatabaseType databaseType) {
		return (databaseType != DatabaseType.MARIADB) ? databaseType.name() : DatabaseType.MYSQL.name();
	}
```
in the other abstract sibling `AbstractJdbcJobSearchableExecutionDaoTests`, do you mind pushing that function up to `AbstractDaoTests` and using it instead?
```java
super.prepareForTest(dbContainer, "mariadb", determineDatabaseType(DatabaseType.MARIADB));
```